components/style/gecko/wrapper.rs

```diff
@@ -614,6 +618,10 @@ impl<'le> TElement for GeckoElement<'le> {
     }
 
     fn get_animation_rules(&self) -> AnimationRules {
+        if !self.may_have_animations() {
+            return AnimationRules(None, None);
```

Comment:
I think both self.get_animation_rule() and self.get_transition_rule() call get_animation_rule(self, cascade_level), which will return None, so we don't need to add this early return. However, it's ok to avoid redundant function calls. Thanks for fixing this.